agent/router/serf_flooder.go

```diff
@@ -93,17 +93,19 @@ func FloodJoins(logger hclog.Logger, addrFn FloodAddrFn, portFn FloodPortFn,
 			}
 		}
 
+		globalServerName := fmt.Sprintf("%s.%s", server.Name, server.Datacenter)
+
 		// Do the join!
-		n, err := globalSerf.Join([]string{addr}, true)
+		n, err := globalSerf.Join([]string{globalServerName + "/" + addr}, true)
```

Comment:
Is this code only called in the Mesh gateway path now? What happens here if memberlist doesn't have its new node names mode enabled which I think is the case when not using mesh gateways right?

Reply:
Understanding join addresses that are name-slash-prefixed is an update to the library that is independent of the toggle flag, so because the library is updated this is fine with or without fed enabled.